app/jobs/backup_code_backfiller_job.rb

```diff
@@ -0,0 +1,47 @@
+class BackupCodeBackfillerJob < ApplicationJob
+  queue_as :low
+
+  # Helper to be run manually in a Rails console once
+  def self.enqueue_all(batch_size: 10_000)
+    Range.new(1, User.last.id).
+      step(batch_size).
+      each do |start|
+        perform_later(start_id: start, count: batch_size)
+      end
+  end
+
+  # Operates on a batch by user, so that we can give a batch
+  # the same salt to make forward lookups more efficient
+  def perform(start_id:, count:)
+    User.
+      where('id >= ?', start_id).
+      limit(count).
+      each do |user|
+        perform_batch(user.backup_code_configurations)
+      rescue => e
```

Comment:
another approach to this would be to do one user per job, so that we can retry easily

However, that's millions of jobs in the queue and I think we can get slight efficiency wins related to job overhead doing it this way (and also have fewer rows in the DB)

Reply:
I think this way is good 👍🏼